articles/search/search-howto-complex-data-types.md

```diff
@@ -244,6 +244,105 @@ To filter on a complex collection field, you can use a **lambda expression** wit
 
 As with top-level simple fields, simple subfields of complex fields can only be included in filters if they have the **filterable** attribute set to `true` in the index definition. For more information, see the [Create Index API reference](/rest/api/searchservice/create-index).
 
+Azure Search has the limitation that the complex objects in the collections across a single document cannot exceed 3000.
+
+Users will encounter the below error during indexing when complex collections exceed the 3000 limit.
+
+“A collection in your document exceeds the maximum elements across all complex collections limit. The document with key '1052' has '4303' objects in collections (JSON arrays). At most '3000' objects are allowed to be in collections across the entire document. Remove objects from collections and try indexing the document again."
+
+In some use cases, we might need to add more than 3000 items to a collection. In those use cases, we can pipe (|) or use any form of delimiter to delimit the values, concatenate them, and store them as a delimited string. There is no limitation on the number of strings stored in an array in Azure Search. Storing these complex values as strings avoids the limitation. The customer needs to validate whether this workaround meets their scenario requirements.
+
+Consider, For example, it wouldn't be possible to use complex types if the "searchScope" array below had more than 3000 elements.
+
+```json
+
+"searchScope": [
+  {
+     "countryCode": "FRA",
+     "productCode": 1234,
+     "categoryCode": "C100" 
+  },
+  {
+     "countryCode": "USA",
+     "productCode": 1235,
+     "categoryCode": "C200" 
+  }
+]
+```
+
+Storing these complex values as strings with a delimiter avoids the limitation
+
+```json
+"searchScope": [
+        "|FRA|1234|C100|",
+        "|FRA|*|*|",
+        "|*|1234|*|",
+        "|*|*|C100|",
+        "|FRA|*|C100|",
+        "|*|1234|C100|"
+]
+
+```
+The reason we have stored the values with wildcards instead of just storing them as below
```

Comment:
Rather than storing these with wildcards, we might want to suggest using a custom analyzer that splits the word on | to cut down on storage size.

https://learn.microsoft.com/en-us/azure/search/index-add-custom-analyzers

Reply:
Have added the custom analyzer as alternative as suggested